src/EventListener/ConditionallyRemoveVaryHeaderListener.php

```diff
@@ -57,10 +57,12 @@ public function onKernelResponse(ResponseEvent $event)
         }
 
         $response = $event->getResponse();
-        $varyHeaders = $response->headers->all('vary');
+        $varyHeaders = array_map('strtolower', $response->headers->all('vary'));
 
         foreach ($this->userIdentifierHeaders as $removableVary) {
-            unset($varyHeaders[array_search(strtolower($removableVary), [$varyHeaders])]);
+            if (($key = array_search($removableVary, $varyHeaders)) !== false) {
```

Comment:
Please, untangle that expression to two lines, it is not very readable (at least to me) in that form.

Reply:
Fixed in 7223b9a464a479efc9165820cf5d9f48efb92199